src/map/skill.c

```diff
@@ -16582,6 +16589,23 @@ static int skill_check_condition_castend(struct map_session_data *sd, uint16 ski
 			break;
 		}
 
+		case MO_ABSORBSPIRITS: {
+			struct map_session_data *tgtsd = BL_CAST(BL_PC, target);
+			if (tgtsd == NULL)
+				break;
+
+			if (tgtsd->spiritball == 0 || (tgtsd->job & MAPID_BASEMASK) == MAPID_GUNSLINGER) {
+				clif->skill_fail(sd, skill_id, USESKILL_FAIL, 0, 0);
+				return 0;
+			}
+
+			if (sd != tgtsd && battle->check_target(&sd->bl, &tgtsd->bl, BCT_NOENEMY) == 1) {
+				clif->skill_fail(sd, skill_id, USESKILL_FAIL, 0, 0);
+				return 0;
+			}
+			break;
+		}
```

Comment:
I am not sure what is the proper way to use blocks around case per Herc's style guide. I couldn't find a mention for this. And this function actually uses 2 different modes. (see NC_MAGICDECOY below vs AM_SPHEREMINE above) I picked based on my tastes / avoiding an extra tab

Reply:
good point! thanks :D